cpp/arcticdb/storage/rocksdb/rocksdb_storage.cpp

```diff
@@ -125,10 +126,18 @@ void RocksDBStorage::do_read(Composite<VariantKey>&& ks, const ReadVisitor& visi
             std::string k_str = to_serialized_key(k);
             std::string value;
             auto s = db_->Get(::rocksdb::ReadOptions(), handle, ::rocksdb::Slice(k_str), &value);
-            util::check(s.ok(), DEFAULT_ROCKSDB_NOT_OK_ERROR + s.ToString());
-            visitor(k, Segment::from_bytes(reinterpret_cast<uint8_t*>(value.data()), value.size()));
+            storage::check<ErrorCode::E_UNEXPECTED_ROCKSDB_ERROR>(s.IsNotFound() || s.ok(), DEFAULT_ROCKSDB_NOT_OK_ERROR + s.ToString());
+            if(s.IsNotFound()) {
+                log::storage().debug("Failed to read segment for key {}", variant_key_view(k));
+                failed_reads.push_back(k);
+            }
+            else {
```

Comment:
(nit) we usually have the else on the line above

Reply:
Fixed in [9fed291](https://github.com/man-group/ArcticDB/pull/1360/commits/9fed291316113f706a553493480ea20ed0937916).